src/db.rs

```diff
@@ -28,16 +28,20 @@ use crate::internal::{iterator::MatchIterator, tag::Tag};
 use crate::package::Package;
 use streaming_iterator::StreamingIterator;
 
+use std::cell::Cell;
 use std::ffi::CString;
 use std::fmt;
+use std::marker::PhantomData;
 use std::os::unix::ffi::OsStrExt;
 use std::path::Path;
 use std::ptr;
 
 static mut LIB_RPM_CONFIGURED: bool = false;
 
 #[derive(Debug, PartialEq, Eq)]
-pub struct Db {}
+pub struct Db {
+    _this_type_is_not_sync: PhantomData<Cell<()>>,
```

Comment:
You can also do:

```rust
impl !Sync for Db {}
```

Reply:
I believe I tried that, and it's an unstable feature behind a feature flag on nightly:
https://doc.rust-lang.org/nomicon/send-and-sync.html
First code sample from the bottom